File: Support/Fuego/Pythia/pythia-0.4/packages/journal/lib/Info.cc
// -*- C++ -*-
//
//--------------------------------------------------------------------------------
//
//                              Michael A.G. Aivazis
//                       California Institute of Technology
//                       (C) 1998-2003  All Rights Reserved
//
// <LicenseText>
//
//--------------------------------------------------------------------------------
//

#include <portinfo>
#include <map>
#include <string>
#include <sstream>

#include "Category.h"
#include "Diagnostic.h"
#include "Info.h"
#include "Index.h"
#include "LocalIndex.h"

using namespace journal;

// statics
static Info::index_t * _index = new LocalIndex(false);

// interface
void Info::newIndex(index_t * newIndex) {
    delete _index;
    _index = newIndex;
    return;
}

// meta-methods
Info::~Info() {}

Info::Info(string_t name):
    Diagnostic("info", name, _index->category(name))
{}

// version
// $Id$

// End of file

File: Support/Fuego/Pythia/pythia-0.5/packages/journal/tests/info.py
#!/usr/bin/env python
#
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                               Michael A.G. Aivazis
#                        California Institute of Technology
#                        (C) 1998-2003 All Rights Reserved
#
#  <LicenseText>
#
#  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#

from __future__ import print_function

if __name__ == "__main__":

    import journal

    print(journal.copyright())

    info = journal.info("info-1")
    debug = journal.debug("debug")
    info = journal.info("info-2")

    print("state of %s(%s): %s" % (info.name, info.facility, info.state))
    info.activate()
    print("state of %s(%s): %s" % (info.name, info.facility, info.state))

    info.log("hello")

    print("info categories:", journal.infoIndex().categories())
    print("debug categories:", journal.debugIndex().categories())


# version
__id__ = "$Id: info.py,v 1.1.1.1 2003/02/16 04:23:02 aivazis Exp $"

#  End of file
